wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -0,0 +1,268 @@
+package org.bitcoins.wallet.internal
+
+import org.bitcoins.wallet.LockedWallet
+import org.bitcoins.core.protocol.transaction.Transaction
+import org.bitcoins.wallet.models._
+import scala.concurrent.Future
+import org.bitcoins.core.protocol.transaction.TransactionOutput
+import org.bitcoins.wallet.api.AddUtxoSuccess
+import org.bitcoins.wallet.api.AddUtxoError
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.util.FutureUtil
+
+/** Provides functionality for processing transactions. This
+  * includes importing UTXOs spent to our wallet, updating
+  * confirmation counts and marking UTXOs as spent when
+  * spending from our wallet
+  */
+private[wallet] trait TransactionProcessing { self: LockedWallet =>
+  /////////////////////
+  // Public facing API
+
+  /** @inheritdoc */
+  override def processTransaction(
+      transaction: Transaction,
+      confirmations: Int): Future[LockedWallet] = {
```

Comment:
Should this return a `TransactionProcessing` instead of a `LockedWallet`?

Reply:
So the intention here is that `ProcessTxResult` is part of the internal API only, and used to for example log the result of processing a TX. An external user shouldn't need to be aware of any of those details. However, it should return `LockedWalletApi` instead of `LockedWallet`. So I changed that.